src/components/MessagesList/MessagesGroup/Message/Message.spec.js

```diff
@@ -0,0 +1,1060 @@
+					},
+					propsData: messageProps,
+				})
+
+				const actionButton = findActionButton(wrapper, 'Delete')
+				expect(actionButton.exists()).toBe(true)
+
+				await actionButton.find('button').trigger('click')
+
+				expect(deleteMessage).toHaveBeenCalledWith(expect.anything(), {
+					message: {
+						token: TOKEN,
+						id: 123,
+					},
+					placeholder: expect.anything(),
+				})
+
+				await wrapper.vm.$nextTick()
+				expect(wrapper.vm.isDeleting).toBe(true)
+				expect(wrapper.find('.icon-loading-small').exists()).toBe(true)
+
+				resolveDeleteMessage(200)
+				// needs two updates...
+				await wrapper.vm.$nextTick()
+				await wrapper.vm.$nextTick()
+
+				expect(wrapper.vm.isDeleting).toBe(false)
+				expect(wrapper.find('.icon-loading-small').exists()).toBe(false)
+			})
+
+			function testDeleteMessageVisible(visible, mockDate, participantType = PARTICIPANT.TYPE.USER) {
+				store = new Vuex.Store(testStoreConfig)
+
+				// need to mock the date to be within 6h
+				if (!mockDate) {
+					mockDate = new Date('2020-05-07 10:00:00')
+				}
+
+				jest.spyOn(global.Date, 'now')
+					.mockImplementation(() => mockDate)
+
+				const wrapper = shallowMount(Message, {
+					localVue,
+					store,
+					stubs: {
+						ActionButton,
+					},
+					mixins: [{
+						computed: {
+							participant: () => {
+								return {
+									actorId: 'user-id-1',
+									actorType: ATTENDEE.ACTOR_TYPE.USERS,
+									participantType,
+								}
+							},
+						},
+					}],
+					propsData: messageProps,
+				})
+
+				const actionButton = findActionButton(wrapper, 'Delete')
+				expect(actionButton.exists()).toBe(visible)
+			}
+
+			test('hides delete action when message is older than 6 hours', () => {
+				testDeleteMessageVisible(false, new Date('2020-05-07 15:24:00'))
+			})
+
+			test('hides delete action when the conversation is read-only', () => {
+				conversationProps.readOnly = CONVERSATION.STATE.READ_ONLY
+				testDeleteMessageVisible(false)
+			})
+
+			test('hides delete action for file messages', () => {
+				messageProps.message = '{file}'
+				messageProps.messageParameters.file = {}
+				testDeleteMessageVisible(false)
+			})
+
+			test('hides delete action on other people messages for non-moderators', () => {
+				messageProps.actorId = 'another-user'
+				conversationProps.type = CONVERSATION.TYPE.GROUP
+				testDeleteMessageVisible(false)
+			})
+
+			test('shows delete action on other people messages for moderators', () => {
+				messageProps.actorId = 'another-user'
+				conversationProps.type = CONVERSATION.TYPE.GROUP
+				testDeleteMessageVisible(true, null, PARTICIPANT.TYPE.MODERATOR)
+			})
+
+			test('shows delete action on other people messages for owner', () => {
+				messageProps.actorId = 'another-user'
+				conversationProps.type = CONVERSATION.TYPE.PUBLIC
+				testDeleteMessageVisible(true, null, PARTICIPANT.TYPE.OWNER)
+			})
+
+			test('does not show delete action even for guest moderators', () => {
```

Comment:
@nickvergessen I assume this is intentional

Reply:
https://github.com/nextcloud/spreed/blob/2c3f80bb08793fec62a95478187ffafffb35bbff/lib/Controller/ChatController.php#L557

Allow guest moderators is false, so yeah only normal moderators/owners